### DISCUSSION TITLE
Low performance when showing menu as party of a table TD

### DISCUSSION THREAD

**Discussion by cubecleveland** [ariakit/ariakit#909] [Questions]
Hello, im using react-table and in conjuction showing a menu item using reakit on every TD in the table.

Noticing quiet a heavy delay when the table is getting rerdred
Iv isoltaed the issue and it seems like the menu component is the culprit.

My menu item is nothing but the example form the website, unstyled ....
To futher isolate the issue i noticed that the problem is the useMenuState() that is being called in the comp

For example: to test ive just called setMenuState() without event rendering a menu item and the lag persissts,,,

Is there anythithing that can be done ? 
FYI, im only rendering 30 TD's -- so 30 menus'....

**Comment by diegohaz** [comment]
It's weird that the state hooks have a performance impact alone without rendering the components. If you can create a simple reproduction of that on CodeSandbox, that will be easier to check.

`useMenuState` is pretty much a combination of [`useCompositeState`](https://reakit.io/docs/composite/) and [`usePopoverState`](https://reakit.io/docs/popover/), so I'd ask you to try these two modules in isolation in the same context to see which one has more impact.

**Reply by cubecleveland** [reply]
Hello,

I have confirmed a few things....

You are correct... simply putting the state there doesnt impact performabnce( thaqt was a ctually my dev tools that broke the performance :) However... renderign the menu... does.

So again if i draw 30 TD's and each one of them has a menu.... it breaks the performance....
Not sure why but at least now it makes sense.

aloso if i only put the button there it does not break performance...
as soon as i add <Menu {...menu} aria-label="Preferences"></Menu> there is a performance break


Question, maybe each menu has t have a seperate identifier... ? not sure if  that makes any sense or if im missing anytthing

**Reply by diegohaz** [reply]
You can try to conditionally render the menu contents. You can use the technique described here: https://reakit.io/docs/disclosure/#conditionally-rendering (Menu uses DisclosureContent underneath).

**Reply by cubecleveland** [reply]
i will give that a try.
FYI, great library.
